src/EnergyPlus/UtilityRoutines.cc

```diff
@@ -810,10 +810,12 @@ namespace UtilityRoutines {
         CloseReportIllumMaps(ioFiles);
         CloseDFSFile(ioFiles);
 
-        if (DebugOutput || ioFiles.debug.position() > 0) {
-            ioFiles.debug.close();
-        } else {
-            ioFiles.debug.del();
+        if (ioFiles.debug.good()) {
+            if (DebugOutput || ioFiles.debug.position() > 0) {
```

Comment:
Really you can just do `if (DebugOutput || (ioFiles.debug.good() && ioFiles.debug.position() > 0)) {`.
`del()` will check if the stream is valid before attempting to use it.
Additionally, I believe it is hard crashing since `position()`  has `noexcept` but it clearly CAN throw an exception so it hard crashes instead (@Myoldmopar @lefticus).